### DISCUSSION TITLE
Disable CTU for boundary driving applications

### DISCUSSION THREAD

**Discussion by kyriakosTapinou** [PrincetonUniversity/athena#618] [General]
I'd like to disable the CTU flux correction step for the left hand reconstruction at a boundary interface but allow the correction for the right hand reconstruction. Is the best way to just overwrite the register with the original values or is there straightforward way to affect the flux correction step for that left side reconstruction value?

**Comment by felker** [comment]
FYI, there is no CTU (corner transport upwind) integrator in Athena++, unlike [in Athena 4.2](https://princetonuniversity.github.io/Athena-Cversion/AthenaDocsUGInt). 

I assume you are asking about CT (constrained transport), given the context, but I don't want to mislead other readers

**Reply by kyriakosTapinou** [reply]
My mistake! I was reading an outdated paper.

**Reply by felker** [reply]
No worries. In case you are interested, here are some relevant notes about flux correction in the Athena++ scheme from Stone et al (2020). The magnetic field fluxes (emf) **are always corrected** at MeshBlock boundaries, whereas the hydro variable fluxes are only exchanged and corrected if SMR or AMR are used. 

> For the face-centered ﬂuxes of the cell-centered conserved variables, this ﬂux-correction step is straightforward (see Figure 6). In 3D calculations at the interface between different levels, one coarse cell abuts four ﬁne cells (two cells in 2D, and one in 1D). The ﬂux used to update the coarse cell on the face that overlaps with the ﬁne cells is simply replaced with the area-weighted sum of the ﬂuxes from these four ﬁne cells. The step makes use of the communication strategy outlined in the previous section for face-centered data.

> Even without mesh refinement, numerical errors can cause a slight mismatch between the EMFs on shared edges between MeshBlocks. With the CT scheme, such errors never disappear once generated. This problem becomes more prominent when more complex grids with nonuniform mesh spacing and/or curvilinear coordinates are in use. Moreover, aggressive compiler (non-ANSI-conformant) optimizations can introduce and exacerbate differences associated with round-off errors. Therefore, the EMF correction step is applied even when mesh refinement is not used. In this case, the EMFs on two shared edges are replaced with the arithmetic average of their values.
